R/textstat_simil.R

```diff
@@ -13,10 +13,11 @@
 #'   \code{"features"} for word/term features.
 #' @param method method the similarity or distance measure to be used; see
 #'   Details.
-#' @param upper whether the upper triangle of the symmetric \eqn{V \times V}
-#'   matrix is recorded. Only used when \code{value = "dist"}.
-#' @param diag whether the diagonal of the distance matrix should be recorded. .
```

Comment:
`as.dist()` can have `diag` but resulting object of `textstat_simil()` should have all the information. I am strongly against making up values for `as.matrix()` and `as.Matrix()` because diagonals are not always 1.0 or 0 in some measures.

Reply:
Hmmm, I thought `simil(A, A)` always 1.0 but it's not for the `method = "faith"`. Because I think it should be, I filed #1675. If that's not right, then we definitely do not want to make up any values.

If we can exclude the diagonals from the output pairs, then we save as many pairs as their are ndoc or nfeat.

Reply:
Solved in 4a052caf, by removing the faith measure. Now all simil(A,A) = 1.0.